biggus/__init__.py

```diff
@@ -955,7 +955,7 @@ def __init__(self, array, axes=None):
             axes = np.arange(array.ndim)[::-1]
         elif len(axes) != array.ndim:
             raise ValueError('Incorrect number of dimensions.')
-        self.axes = list(axes)
+        self.axes = axes
```

Comment:
@pelson I thought this was fixing the problem at its source. Casting to a list here ensured that `self.axes` was always mutable. Or do you consider it best practice that `self.axes` should honour and maintain what the user provided?